libs/tree-sitter-wing/grammar.js

```diff
@@ -334,7 +353,15 @@ module.exports = grammar({
         optional(
           seq(
             "catch",
-            optional(field("exception_identifier", $.identifier)),
+            optional(
```

Comment:
This is extra grammar just for more specific errors. I guess this is ok, and it's probably the easiest way to deal with this.
I wonder if this is another hint that we need to eventually move to our own parser? @MarkMcCulloh I wonder what's your take on this?

Reply:
I'm definitely in favor of us doing our own parser for error-correcting cases like this and others.

> This is extra grammar just for more specific errors

The way I see it: If we were to write our own parser, we would definitely add code to specifically detect this case. Adding it to our ts grammar is currently our only way to do that.

It's worth noting we have the same issue in several of our syntaxes wrt to parens:

```wing
for (x in [2]) {}
if (let x = 2) {}
```

I think our parser should be able to avoid exploding for all of these